### PR TITLE
fetching, SQS polling, and uploading to S3 are done in batch

### PR DESCRIPTION
**Pull request by chrome99**
- Requests to S3 and fetch requests are done inside Promise.allSettled()
- Lambda function sends messages in batches of 10 using SendMessageBatchCommand
- Fetching is done with timeout
- Output files are named with uniqid, not by url (commented out)

### CODE REVIEW

File: src/consumer/index.js [hunk 1]
```diff
@@ -9,15 +9,45 @@ import { writeOutput } from '../utils/writeOutput.js'
...
 
-  // Write the output to S3
-  await writeOutput(message, body)
+    return messages;
+  } catch (error) {
+    console.error('An error occurred:', error)
+  }
 }
+
+async function fetchWithTimeout(url, timeout) {
```
Comment (agamm) [inline src/consumer/index.js:45]: Maybe move this under utils/consumer.js?

File: src/consumer/index.js [hunk 2]
```diff
@@ -9,15 +9,45 @@ import { writeOutput } from '../utils/writeOutput.js'
...
+    })
+    .map(res => res.value);
 
-  // (TODO) Add your processing logic here...
+    // upload to s3 bucket
+    await writeOutput(okResponses);
 
-  // Write the output to S3
-  await writeOutput(message, body)
+    return messages;
```
Comment (agamm) [inline src/consumer/index.js:39]: Who is getting this?

File: src/consumer/index.js [hunk 3]
```diff
@@ -9,15 +9,45 @@ import { writeOutput } from '../utils/writeOutput.js'
  * @param {object} message The SQS message body as a JSON object
  * @return {void}
  */
-export async function handler(message) {
-  // Make the request
-  const response = await fetch(message.url)
+export async function handler(messages) {
```
Comment (agamm) [inline src/consumer/index.js:12]: Add a comment here explaining what this does and how this is only an example.

**Review by agamm — CHANGES_REQUESTED** [review]
Some comments :)

File: src/consumer/index.js (hunk 2 above)
Comment (chrome99) [reply src/consumer/index.js:39]: My bad, I thought messages are deleted only if you return them, But this isn't true. I Removed this

File: src/consumer/index.js [hunk 4]
```diff
@@ -1,5 +1,6 @@
 import fetch from 'node-fetch'
 import { writeOutput } from '../utils/writeOutput.js'
+import { fetchWithTimeout } from '../utils/consumer.js'
 
 /**
  * This script is the handler that performs any manipulation of the message
```
Comment (agamm) [inline src/consumer/index.js:6]:
```suggestion
 * This is just an example that sends an HTTP request, you can change it to anything you want ;)
```

File: src/consumer/index.js [hunk 5]
```diff
@@ -9,15 +10,39 @@ import { writeOutput } from '../utils/writeOutput.js'
...
+const FETCH_TIMEOUT = 10000;
 
-  // Get the response body as JSON
-  const body = await response.json()
+export async function handler(messages) {
+  // example of handler function that fetches 10 urls
+  try {
+    // running multiple asynchronous tasks
+    const responses = await Promise.allSettled(
+      messages.map(async message => {
```
Comment (agamm) [inline src/consumer/index.js:20]: add a comment that the messages come from the producer ;)

File: src/producer/index.js [hunk 6]
```diff
@@ -10,11 +11,23 @@ import data from './urls.json' assert { type: 'json' }
  */
 export async function handler(event, context) {
   // Send the urls to the SQS queue
```
Comment (agamm) [inline src/producer/index.js:13]: Write something that this sends them batched, and why you are batching them.

**Review by agamm — CHANGES_REQUESTED** [review]
A few commets, but looks better.

**Review by agamm — APPROVED** [review]
Looks good to me!

**Pull request merged**